dockstore-client/src/main/java/io/dockstore/client/cli/nested/WorkflowClient.java

```diff
@@ -466,16 +466,19 @@ public void launchWithArgs(final String entry, final String localEntry, final St
                         final String descriptor = descriptorType.getValue().toLowerCase();
                         LanguageClientInterface languageClientInterface = convertCLIStringToEnum(descriptor);
                         DescriptorLanguage language = DescriptorLanguage.convertShortStringToEnum(descriptor);
-
+                        if (workflow.getDefaultTestParameterFilePath() == null) {
```

Comment:
If I understand this block correctly, if there isn't a default test parameter file, and one wasn't passed in via the CLI, an error is thrown. This means that if a user has a workflow that doesn't take any inputs, they'll still need to pass in a dummy `--json` or `--yaml` parameter, correct?

For example, launching something like https://dockstore.org/workflows/github.com/dockstore-testing/dockstore-whalesay2:master?tab=info, should be launch-able without any default or explicit parameter files:
![Screen Shot 2022-05-20 at 10 28 09 AM](https://user-images.githubusercontent.com/36607471/169549394-a660b940-2c58-483a-9973-3c196fbe71cf.png)

Reply:
@Richard-Hansen 
Yeah the idea was that if a workflow does not specify a default test parameter file, the user should specify one.

Is there an attribute to a workflow that can tell me if there needs to be a test input file? That would probably be the best way to decide if parameter is required.

Reply:
@coverbeck @denis-yuen do you think its worth inspecting if there is a default test parameter files specified in the workflow, I was trying to avoid users running into an error if there was no default specified but was required. If this is a non-issue I'd prefer to remove the check.

Reply:
TLDR; I think don't bother.

You wouldn't use `Workflow.getDefaultTestParameterFilePath` for this. It's a hint for Dockstore indexing the workflow; it tells Dockstore where to look for a test parameter file when creating a new version via refresh. For example,

1. For the repo foo, the user sets the default test parameter file path  to `/goo.json`.
2. The user adds a new branch `bar` in Git
3. The user clicks refresh in Dockstore for the repo foo
4. Dockstore reads the repo foo, looking for the file /goo.json in Git. If it finds it, /goo.json and its contents get added to the version `bar` on Dockstore. If it doesn't find it, then it doesn't get added.

Finally, whether or not `goo.json` gets added to the version on Dockstore not, it does not indicate whether the workflow/tool **requires** inputs or not.

To do that, you would need to do something like what the [convert](https://github.com/dockstore/dockstore-cli/blob/b7cb750e99700885b009a401e99ea98d5dceced5/dockstore-client/src/main/java/io/dockstore/client/cli/nested/AbstractEntryClient.java#L620) method does. And even then, I think in theory it could return only optional parameters, so it wouldn't be 100% reliable.

Just let the workflow engine complain when running the workflow, IMHO.